bundles/org.openhab.binding.casokitchen/src/main/java/org/openhab/binding/casokitchen/internal/CasoKitchenHandlerFactory.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.casokitchen.internal;
+
+import static org.openhab.binding.casokitchen.internal.CasoKitchenBindingConstants.THING_TYPE_WINECOOLER;
+
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.casokitchen.internal.handler.TwoZonesWinecoolerHandler;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link CasoKitchenHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.casokitchen", service = ThingHandlerFactory.class)
+public class CasoKitchenHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_WINECOOLER);
+    private HttpClient httpClient;
+    private TimeZoneProvider timeZoneProvider;
+
+    @Activate
+    public CasoKitchenHandlerFactory(final @Reference HttpClientFactory httpClientFactory,
+            final @Reference TimeZoneProvider tzp) {
+        httpClient = httpClientFactory.getCommonHttpClient();
```

Comment:
Please store the factory and create a httpclient for each time a thing handler is created.

Reply:
Changed